backend/app/adapter/facebook/identityprovider.go

```diff
@@ -37,13 +29,24 @@ type IdentityProvider struct {
 
 // GetAuthorizationURL retrieves the URL of Facebook sign in page.
 func (g IdentityProvider) GetAuthorizationURL() string {
-	escapedScope := url.QueryEscape(fbScopes)
 	clientID := g.clientID
 	redirectURI := g.redirectURI
 	responseType := fbResponseType
+	scope := fbScopes
+
+	u, err := url.Parse(fbAuthorizationAPI)
+	if err != nil {
+		return ""
+	}
+
+	query := u.Query()
+	query.Set("client_id", clientID)
+	query.Set("redirect_uri", redirectURI)
+	query.Set("scope", scope)
+	query.Set("response_type", responseType)
+	u.RawQuery = query.Encode()
 
-	return fmt.Sprintf("%s?client_id=%s&redirect_uri=%s&scope=%s&response_type=%s",
```

Comment:
Well done! Awesome work!

Reply:
I will add integration test in another pr

Reply:
@dawntomm Let's include necessary automated tests in this PR so that we won't forget!

Reply:
@dawntomm I think we should add the necessary tests in this PR. It is quite difficult to find a code error by review...

Reply:
will do